tsdb/querier.go

```diff
@@ -509,9 +499,11 @@ func (s *mergedSeriesSet) nextAll() {
 	for _, ss := range s.all {
 		if ss.Next() {
 			s.buf = append(s.buf, ss)
+			s.ws = append(s.ws, ss.Warnings()...)
 		} else if ss.Err() != nil {
```

Comment:
I would vote for no else but `continue` instead (: https://thanos.io/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else